src/net.cpp

```diff
@@ -1404,6 +1404,7 @@ void CConnman::CalculateNumConnectionsChangedStats()
         LOCK(pnode->cs_vRecv);
         for (const mapMsgCmdSize::value_type &i : pnode->mapRecvBytesPerMsgCmd)
             mapRecvBytesMsgStats[i.first] += i.second;
+        LOCK(pnode->cs_vSend);
         for (const mapMsgCmdSize::value_type &i : pnode->mapSendBytesPerMsgCmd)
             mapSentBytesMsgStats[i.first] += i.second;
```

Comment:
20816: could reduce the scope of locks here

Reply:
Hello @UdjinM6 requesting suggestion for apt place for LOCK

Reply:
```diff
diff --git a/src/net.cpp b/src/net.cpp
index a7d3c4183f..c85ae2eda7 100644
--- a/src/net.cpp
+++ b/src/net.cpp
@@ -1401,12 +1401,16 @@ void CConnman::CalculateNumConnectionsChangedStats()
     mapSentBytesMsgStats[NET_MESSAGE_COMMAND_OTHER] = 0;
     auto vNodesCopy = CopyNodeVector(CConnman::FullyConnectedOnly);
     for (auto pnode : vNodesCopy) {
-        LOCK(pnode->cs_vRecv);
-        for (const mapMsgCmdSize::value_type &i : pnode->mapRecvBytesPerMsgCmd)
-            mapRecvBytesMsgStats[i.first] += i.second;
-        LOCK(pnode->cs_vSend);
-        for (const mapMsgCmdSize::value_type &i : pnode->mapSendBytesPerMsgCmd)
-            mapSentBytesMsgStats[i.first] += i.second;
+        {
+            LOCK(pnode->cs_vRecv);
+            for (const mapMsgCmdSize::value_type &i : pnode->mapRecvBytesPerMsgCmd)
+                mapRecvBytesMsgStats[i.first] += i.second;
+        }
+        {
+            LOCK(pnode->cs_vSend);
+            for (const mapMsgCmdSize::value_type &i : pnode->mapSendBytesPerMsgCmd)
+                mapSentBytesMsgStats[i.first] += i.second;
+        }
         if(pnode->fClient)
             spvNodes++;
         else
```